### PR TITLE
Fixes #15608: When creating a rule via the API the id parameter is only read from url parameters, and not from json

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/15608

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala [hunk 1]
```diff
@@ -126,18 +126,19 @@ class RuleApi(
...
-          case Full(arg) =>
-            val restRule = restExtractor.extractRuleFromJSON(arg)
-            apiV2.createRule(restRule,req)
-          case eb:EmptyBox=>
-            toJsonError(None, JString("No Json data sent"))("createRule",restExtractor.extractPrettify(req.params))
-        }
-      } else {
-        val restRule = restExtractor.extractRule(req.params)
-        apiV2.createRule(restRule, req)
+      implicit var action = "createRule"
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala]: no `implicit var` please, each one is terrible for understanding, both together... 
Please only `implicit val`

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala [hunk 2]
```diff
@@ -126,18 +126,19 @@ class RuleApi(
...
+        ruleId <- restExtractor.extractId(req)(x => Full(RuleId(x))).map(_.getOrElse(RuleId(uuidGen.newUuid)))
+        optCloneId <- restExtractor.extractString("source")(req)(x => Full(RuleId(x)))
+        result <- apiV2.createRule(restRule, ruleId, optCloneId, authzToken.actor)
+      } yield {
+        if (optCloneId.nonEmpty)
+          action = "cloneRule"
+        result
       }
+
+      actionResponse(response, req, "Could not create Rule", None, authzToken.actor)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala]: Why `None` here? Shouldn't it depend of the kind of response?

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**